backend/src/auth/interface.ts

```diff
@@ -41,6 +52,7 @@ interface ResourceAccess {
 }
 
 export interface RequestWithRoles extends Request {
-  role: Role;
+  role?: Role;
+  program: Program[];
```

Comment:
Why array, given that `admin` is used for both?

Reply:
Why, indeed? Will change that, it does not need to be an array